code/modules/mob/living/living.dm

```diff
@@ -443,6 +443,12 @@
 	if (s_active && !(s_active.ClickAccessible(src, depth=STORAGE_VIEW_DEPTH) || s_active.Adjacent(src)))
 		s_active.close(src)
 
+	for(var/mob/M in oview(src))
+		if(M.client && (src in cone(M, OPPOSITE_DIR(M.dir), view(10, M))))
```

Comment:
the way this works, you could refactor it to cache the details for cone and view() by pivoting it to src rather then m so it only needs to be done once.

I don't know of any situation for something to be in one things view but for that thing to not be in its view, view is two way link is it not?